codebase/src/cpp/ieee1516e/src/types/time/HLAlogicalTimeFactoryFactory.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ *   Copyright 2012 The Portico Project
+ *
+ *   This file is part of portico.
+ *
+ *   portico is free software; you can redistribute it and/or modify
+ *   it under the terms of the Common Developer and Distribution License (CDDL)
+ *   as published by Sun Microsystems. For more information see the LICENSE file.
+ *
+ *   Use of this software is strictly AT YOUR OWN RISK!!!
+ *   If something bad happens you do not have permission to come crying to me.
+ *   (that goes for your lawyer as well)
+ *
+ */
+#include "common.h"
+
+IEEE1516E_NS_START
+
+//------------------------------------------------------------------------------------------
+//                                      STATIC METHODS
+//------------------------------------------------------------------------------------------
+/*
+ * Provides a factory for the standard logical time types HLAfloat64Time
+ * and HLAinteger64Time. The RTI reference time library's LogicalTimeFactoryFactory
+ * should just forward requests to here.
+ */
+std::auto_ptr<LogicalTimeFactory>
+HLAlogicalTimeFactoryFactory::makeLogicalTimeFactory( const std::wstring& implementationName )
```

Comment:
Note this method is declared `RTI_EXPORT` in the standard headers, thus it must be compiled within the RTI DLL. If we compile it in the fedtime DLL project then `RTI_EXPORT` isn't defined and we get inconsistent DLL linkage errors.